python/lsst/afw/cameraGeom/camera.py

```diff
@@ -106,8 +129,38 @@ def findDetectors(self, cameraPoint):
                 detectorList.append(detector)
         return detectorList
 
+    def findDetectorsArray(self, xCoord, yCoord, coordSys):
+        """!Find the detectors that cover a list of points specified by x and y coordinates in any system
+
+        @param[in] xCoord is an array of x coordinates
+        @param[in] yCoord is an array of y coordinates
+        @param[in] coordSys is the coordinate system in which xCoord and yCoord are defined
+        @return a list of lists; each list contains the names of all detectors which contain the
+        corresponding point
+        """
+
```

Comment:
I think there are three different issues in this discussion and I fear they are getting conflated:

1) The speedup. This is caused by using findDetectorsArray instead of findDetectors, and it makes sense because you are vectorizing coordinate transformations. That is perfectly understandable and does not require a ticket.

2) findDetectorsArray x,y arrays vs a list of Point2D. The first thing findDetectorsArray does is turn the x and y array into a list of findDetectorsArray (in python). All I'm asking is that the caller do this, instead, so findDetectorsArray can have a cleaner and API that looks more like the API of TransformMap.

3) Eventually we will want a way of passing around lists of points and possibly coordinates (e.g. ICRS) as numpy arrays. One obvious representation of a list of points in numpy is a structured array with x and y fields. But how do we get that into C++ (e.g. for the vector version of TransformMap.transform)? Coordinates add additional complications. This issue could probably use a separate ticket; it certainly should not be solved on this one.